pkg/messaging/v1/replayable_request_test.go

```diff
@@ -215,3 +217,29 @@ func TestReplayableRequest(t *testing.T) {
 		})
 	})
 }
+
+// readerCloser is a io.Reader that can be closed. Once the stream is closed, reading from it returns an error.
+type readerCloser struct {
```

Comment:
This is added here to simulate reading from a `http.Response.Body` which has already been closed.